unitycatalog/functions/databricks.py

```diff
@@ -0,0 +1,549 @@
+import datetime
+import decimal
+import inspect
+import json
+import logging
+import re
+import time
+from dataclasses import dataclass
+from enum import Enum
+from io import StringIO
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, Union
+
+from typing_extensions import override
+
+from unitycatalog.functions.client import BaseFunctionClient, FunctionExecutionResult
+
+if TYPE_CHECKING:
+    from databricks.sdk import WorkspaceClient
+    from databricks.sdk.service.catalog import (
+        ColumnTypeName,
+        CreateFunction,
+        FunctionInfo,
+        FunctionParameterInfo,
+    )
+    from databricks.sdk.service.sql import StatementParameterListItem
+
+EXECUTE_FUNCTION_ARG_NAME = "__execution_args__"
+DEFAULT_EXECUTE_FUNCTION_ARGS = {
+    "wait_timeout": "30s",
+    "row_limit": 100,
+    "byte_limit": 4096,
+}
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_databricks_workspace_client() -> "WorkspaceClient":
+    try:
+        from databricks.sdk import WorkspaceClient
+    except ImportError as e:
+        raise ImportError(
+            "Could not import databricks-sdk python package. "
+            "If you want to use databricks backend then "
+            "please install it with `pip install databricks-sdk`."
+        ) from e
+    return WorkspaceClient()
+
+
+def column_type_to_python_type(column_type: Enum) -> Any:
+    from databricks.sdk.service.catalog import ColumnTypeName
```

Comment:
Is there a similar mechanism for the OSS UC catalog to retrieve the type conversion fields? If so, should we externalize an abstract implementation of this into a common util module?

Reply:
Make sense! OSS UC uses Literal["Boolean", ...] with the same value as this mapping here, this is Enum so we can also get the string value!